napytau/gui/components/graph.py

```diff
@@ -28,57 +40,171 @@ def update_plot(self) -> None:
             row=0, column=0, rowspan=2, padx=10, pady=10, sticky="nsew"
         )
         self.graph_frame.grid_propagate(False)
 
-    def plot(self, value: int, appearance: str) -> Canvas:
+    def plot(self, appearance: str) -> Canvas:
         """
         Plot the graph.
         :param appearance: The appearance mode.
-        :param value: The value.
         :return: The canvas.
         """
-        # the figure that will contain the plot
-        fig = Figure(figsize=(3, 2), dpi=100, facecolor="white", edgecolor="black")
-
-        # setting color in dependence of appearance mode
-        if appearance == "Light":
-            main_color = "white"
-            secondary_color = "#000000"
-        else:
-            main_color = "#151515"
-            secondary_color = "#ffffff"
 
-        fig.patch.set_facecolor(main_color)
+        # Placeholder datapoints:
+        y_data = [2, 4, 5, 7, 8, 9, 11, 15]
+        distances = [1, 2, 3, 4, 5, 6, 7, 8]
+        x_data = [0, 1, 2, 3, 4, 5, 6, 7]
 
-        # list of squares
-        y = [(i - 50) ** value for i in range(101)]
+        # the figure that will contain the plot
+        fig = Figure(figsize=(3, 2), dpi=100, facecolor=Color.WHITE, edgecolor=Color.BLACK)
 
         # adding the subplot
-        plot1 = fig.add_subplot(111)
+        Axes_1 = fig.add_subplot(111)
 
-        # set color of background
-        plot1.set_facecolor(main_color)
+        # set colors according to appearance mode
+        self.set_colors(appearance)
 
-        # set color of ticks
-        plot1.tick_params(axis="x", colors=secondary_color)
-        plot1.tick_params(axis="y", colors=secondary_color)
+        # apply colors onto figure and axes
+        self.apply_coloring(fig, Axes_1)
 
         # add grid style
-        plot1.grid(
-            True, which="both", color=secondary_color, linestyle="--", linewidth=0.3
+        Axes_1.grid(
+            True, which="both", color=self.secondary_color, linestyle="--", linewidth=0.3
         )
 
-        # plotting the graph
-        plot1.plot(y)
+        # draw the markers on the axes
+        self.plot_markers(x_data, y_data, distances, Axes_1)
+
+        # draw the fitting curve on the axes
+        self.plot_fitting_curve(x_data, y_data, Axes_1)
+
 
         # creating the Tkinter canvas
         # containing the Matplotlib figure
         canvas = FigureCanvasTkAgg(fig, master=self.parent)
         canvas.draw()
 
         return canvas.get_tk_widget()
+
+    def set_colors(self, appearance: str) -> None:
+
+        if appearance == "Light":
+            self.main_color = Color.WHITE
+            self.secondary_color = Color.BLACK
+            self.main_marker_color = Color.DARK_GREEN
+            #.
+            #.
+            #.
+        else:
+            self.main_color = Color.DARK_GRAY
+            self.secondary_color = Color.WHITE
+            self.main_marker_color = Color.LIGHT_GREEN
+            #.
+            #.
+            #.
+
```

Comment:
I don't think these comments are needed.

Reply:
They will be removed later on. My intention was to show more clearly that this function can be expanded.